### PR TITLE
Rust: Tags Refactor

### PR DESCRIPTION
**Pull request by droogmic**

### CODE REVIEW

**Comment by droogmic** [comment]
I had to redo the CI again... still not happy with it :)

**Comment by droogmic** [comment]
> I had to redo the CI again... still not happy with it :)

(it was not reporting errors...)

File: .githooks/check [hunk 1]
```diff
@@ -1,26 +1,38 @@
-#!/bin/sh
+#!/bin/env bash
```
Comment (dabreegster) [inline .githooks/check:1]: I feel like we're reinventing test runners :(  But I don't know a simpler approach. Shell not propagating errors nicely despite `set -e` is definitely frustrating

**Review by dabreegster — APPROVED** [review]
If it's not loads of trouble next time, splitting up refactoring and CI fixes would be preferable. Fine here, but if it's ever a big change / heavy feature, a focused diff can be useful.

Thanks for all of the work so far!

**Comment by droogmic** [comment]
> If it's not loads of trouble next time, splitting up refactoring and CI fixes would be preferable. Fine here, but if it's ever a big change / heavy feature, a focused diff can be useful.
> 
> Thanks for all of the work so far!

agreed, laziness strikes again

File: .githooks/check (hunk 1 above)
Comment (droogmic) [reply .githooks/check:1]: tbh, I am not liking my refactoring.

I might revert the effort I put into making the githooks and actions share the same source of truth. I just don't think it is worth the effort :(

**Pull request merged**